From 7578a75ab99e39c02639178778015f29f6dc5142 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@foss.st.com>
Date: Mon, 3 Mar 2025 11:21:06 +0100
Subject: [PATCH] arm64: dts: st: update USART1 DMA mechanism on
 stm32mp211.dtsi

To fix issue during Bluetooth loading Firmware :
[   16.029919] Bluetooth: hci0: BCM: chip id 135
[   16.030514] Bluetooth: hci0: BCM: features 0x0f
[   18.053146] Bluetooth: hci0: command 0x0c14 tx timeout
[   18.053232] Bluetooth: hci0: BCM: Reading local name failed (-110)

Change-Id: I591bf91507b6619f472620b81250d43d2f2d8fc9
Signed-off-by: Christophe Roullier <christophe.roullier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/441099
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp211.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp211.dtsi b/arch/arm64/boot/dts/st/stm32mp211.dtsi
index 94570804e5c6..a2a886976f75 100644
--- a/arch/arm64/boot/dts/st/stm32mp211.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp211.dtsi
@@ -1332,7 +1332,7 @@ usart1: serial@40330000 {
 				reg = <0x40330000 0x0 0x400>;
 				interrupts-extended = <&exti1 26 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART1>;
-				dmas = <&hpdma 9 0x40 0x12>,
+				dmas = <&hpdma 9 0x40 0x10012>,
 				       <&hpdma 10 0x40 0x3021>;
 				dma-names = "rx", "tx";
 				access-controllers = <&rifsc 31>;
-- 
2.50.1

